### ISSUE TITLE
Running --auto-version when building a release always breaks (Ubuntu 18.04 LTS, Elixir 1.10.1)

### ISSUE THREAD

**Issue by inooid** [edeliver/edeliver#324]
Running `mix edeliver build release --auto-version=commit-count` always runs breaks no matter what type of `auto-version` you use. I can build a release without auto-version, but as soon as I add auto version it breaks.

I've tested both on my deploy server as well as a locally built docker container (using Ubuntu 18.04 LTS) and could reproduce on both. I also created the smallest app: https://github.com/inooid/elixir-edeliver-auto-version. I am still looking into sharing the docker image as well, but for now I just used a stock `ubuntu` image with ssh access that I could use.

### Environment

* Edeliver version (`mix edeliver --version`): `edeliver v1.6.1 | https://github.com/edeliver/edeliver`
* Elixir version (`elixir -v`): `Elixir 1.10.1 (compiled with Erlang/OTP 21)`
* Build tool and version (distillery): `distillery 2.1.1` and `mix`
* Operating system (on build / deploy hosts): `build: Ubuntu 18.04 LTS` same for deploy, but it doesn't get there
* Are you using an umbrella project (yes|no): no

### Verbose output
```shell
23:54:05.393 [error] GenServer Mix.ProjectStack terminating
** (FunctionClauseError) no function clause matching in Mix.ProjectStack.handle_call/3
    (mix 1.10.1) lib/mix/project_stack.ex:262: Mix.ProjectStack.handle_call({:get_and_update, #Function<13.89951273/1 in Mix.Tasks.Release.Version.update_release_version/2>}, {#PID<0.92.0>, #Reference<0.753078404.1789657090.72962>}, {[%{after_compiler: %{}, compile_env: :unset, config: [aliases: [], build_embedded: false, build_per_environment: true, build_scm: Mix.SCM.Path, config_path: "config/config.exs", consolidate_protocols: true, default_task: "run", deps_path: "deps", elixirc_paths: ["lib"], erlc_paths: ["src"], erlc_include_path: "include", erlc_options: [], lockfile: "mix.lock", preferred_cli_env: [], app: :edeliver_test, version: "0.1.0", elixir: "~> 1.10", start_permanent: true, deps: [{:edeliver, ">= 1.6.0"}, {:distillery, "~> 2.0", [warn_missing: false]}]], config_apps: [], config_files: ["/tmp/edeliver/edeliver_test/builds/config/config.exs", "/tmp/edeliver/edeliver_test/builds/mix.exs"], config_mtime: nil, file: "/tmp/edeliver/edeliver_test/builds/mix.exs", io_done: false, name: EdeliverTest.MixProject, pos: 0, recursing?: false}], []})
    (stdlib 3.11.2) gen_server.erl:661: :gen_server.try_handle_call/4
    (stdlib 3.11.2) gen_server.erl:690: :gen_server.handle_msg/6
    (stdlib 3.11.2) proc_lib.erl:249: :proc_lib.init_p_do_apply/3
Last message (from #PID<0.92.0>): {:get_and_update, #Function<13.89951273/1 in Mix.Tasks.Release.Version.update_release_version/2>}
State: {[%{after_compiler: %{}, compile_env: :unset, config: [aliases: [], build_embedded: false, build_per_environment: true, build_scm: Mix.SCM.Path, config_path: "config/config.exs", consolidate_protocols: true, default_task: "run", deps_path: "deps", elixirc_paths: ["lib"], erlc_paths: ["src"], erlc_include_path: "include", erlc_options: [], lockfile: "mix.lock", preferred_cli_env: [], app: :edeliver_test, version: "0.1.0", elixir: "~> 1.10", start_permanent: true, deps: [{:edeliver, ">= 1.6.0"}, {:distillery, "~> 2.0", [warn_missing: false]}]], config_apps: [], config_files: ["/tmp/edeliver/edeliver_test/builds/config/config.exs", "/tmp/edeliver/edeliver_test/builds/mix.exs"], config_mtime: nil, file: "/tmp/edeliver/edeliver_test/builds/mix.exs", io_done: false, name: EdeliverTest.MixProject, pos: 0, recursing?: false}], []}
Client #PID<0.92.0> is alive

    (stdlib 3.11.2) gen.erl:167: :gen.do_call/4
    (elixir 1.10.1) lib/gen_server.ex:1020: GenServer.call/3
    lib/mix/tasks/release_version.ex:87: Mix.Tasks.Release.Version.update_release_version/2
    (mix 1.10.1) lib/mix/task.ex:330: Mix.Task.run_task/3
    (elixir 1.10.1) lib/enum.ex:783: Enum."-each/2-lists^foreach/1-0-"/2
    (elixir 1.10.1) lib/enum.ex:783: Enum.each/2
    (mix 1.10.1) lib/mix/task.ex:330: Mix.Task.run_task/3
    (mix 1.10.1) lib/mix/cli.ex:82: Mix.CLI.run_task/2

23:54:05.395 [info]  Application mix exited: shutdown
** (exit) exited in: GenServer.call(Mix.ProjectStack, {:get_and_update, #Function<13.89951273/1 in Mix.Tasks.Release.Version.update_release_version/2>}, 5000)
    ** (EXIT) an exception was raised:
        ** (FunctionClauseError) no function clause matching in Mix.ProjectStack.handle_call/3
            (mix 1.10.1) lib/mix/project_stack.ex:262: Mix.ProjectStack.handle_call({:get_and_update, #Function<13.89951273/1 in Mix.Tasks.Release.Version.update_release_version/2>}, {#PID<0.92.0>, #Reference<0.753078404.1789657090.72962>}, {[%{after_compiler: %{}, compile_env: :unset, config: [aliases: [], build_embedded: false, build_per_environment: true, build_scm: Mix.SCM.Path, config_path: "config/config.exs", consolidate_protocols: true, default_task: "run", deps_path: "deps", elixirc_paths: ["lib"], erlc_paths: ["src"], erlc_include_path: "include", erlc_options: [], lockfile: "mix.lock", preferred_cli_env: [], app: :edeliver_test, version: "0.1.0", elixir: "~> 1.10", start_permanent: true, deps: [{:edeliver, ">= 1.6.0"}, {:distillery, "~> 2.0", [warn_missing: false]}]], config_apps: [], config_files: ["/tmp/edeliver/edeliver_test/builds/config/config.exs", "/tmp/edeliver/edeliver_test/builds/mix.exs"], config_mtime: nil, file: "/tmp/edeliver/edeliver_test/builds/mix.exs", io_done: false, name: EdeliverTest.MixProject, pos: 0, recursing?: false}], []})
            (stdlib 3.11.2) gen_server.erl:661: :gen_server.try_handle_call/4
            (stdlib 3.11.2) gen_server.erl:690: :gen_server.handle_msg/6
            (stdlib 3.11.2) proc_lib.erl:249: :proc_lib.init_p_do_apply/3
    (elixir 1.10.1) lib/gen_server.ex:1023: GenServer.call/3
    lib/mix/tasks/release_version.ex:87: Mix.Tasks.Release.Version.update_release_version/2
    (mix 1.10.1) lib/mix/task.ex:330: Mix.Task.run_task/3
    (elixir 1.10.1) lib/enum.ex:783: Enum."-each/2-lists^foreach/1-0-"/2
    (elixir 1.10.1) lib/enum.ex:783: Enum.each/2
    (mix 1.10.1) lib/mix/task.ex:330: Mix.Task.run_task/3
    (mix 1.10.1) lib/mix/cli.ex:82: Mix.CLI.run_task/2

A remote command failed on:

  root@local-docker

Output of the command is shown above and the command executed
on that host is printed below for debugging purposes:

FAILED with exit status 1:

    [ -f ~/.profile ] && source ~/.profile
    set -e
    cd /tmp/edeliver/edeliver_test/builds
    if [ "mix" = "rebar" ]; then
      echo "using rebar to compile files"
      [[ "" != "true" ]] && ./rebar  clean skip_deps=true || :
      ./rebar  compile
    elif [ "mix" = "mix" ] && [ "mix" = "mix" ]; then
      echo "Checking whether deps must be compiled for mix version 1.3.[01234]"
      # see https://github.com/boldpoker/edeliver/issues/94
      if mix --version | grep 'Mix 1.3.[01234]' >/dev/null 2>&1 ; then
        echo "Compiling deps because mix version 1.3.[01234] is used"
        APP="edeliver_test" MIX_ENV="prod" mix deps.compile
      fi
      if [[ "" = "true" ]]; then
        APP="edeliver_test" MIX_ENV="prod" AUTO_VERSION="" BRANCH="master" SKIP_RELUP_MODIFICATIONS="" RELUP_MODIFICATION_MODULE="" USING_DISTILLERY="true" mix do release.version commit-count --verbose, compile
      else
        APP="edeliver_test" MIX_ENV="prod" AUTO_VERSION="" BRANCH="master" SKIP_RELUP_MODIFICATIONS="" RELUP_MODIFICATION_MODULE="" USING_DISTILLERY="true" mix do clean, release.version commit-count --verbose, compile
      fi
    elif [ "mix" = "mix" ]; then
      echo "using mix to compile files"
      if [[ "" = "true" ]]; then
        if [[ -n "" ]]; then
          hint_message 'Using --auto-version together with --skip-mix-clean would not work!'
        fi
        APP="edeliver_test" MIX_ENV="prod" AUTO_VERSION="" BRANCH="master" SKIP_RELUP_MODIFICATIONS="" RELUP_MODIFICATION_MODULE="" USING_DISTILLERY="true" mix do deps.compile, release.version commit-count --verbose, compile
      else
        APP="edeliver_test" MIX_ENV="prod" AUTO_VERSION="" BRANCH="master" SKIP_RELUP_MODIFICATIONS="" RELUP_MODIFICATION_MODULE="" USING_DISTILLERY="true" mix do clean, deps.compile, release.version commit-count --verbose, compile
      fi
    fi
```

### Current behavior
Running: `mix edeliver build release --auto-version=commit-count` will break during `-----> Compiling sources`
Running `mix edeliver build release` without `--auto-version` does work normally

### Expected behavior
Running: `mix edeliver build release --auto-version=commit-count` will just finish the build

**Comment by inooid** [comment]
Just tried running it on `Elixir v1.9.4` and it is working there. So it seems to be an `Elixir 1.10` issue!

![image](https://user-images.githubusercontent.com/1291263/74199729-461dfe00-4c65-11ea-9796-e725f6b03664.png)

**Comment by sneako** [comment]
Edeliver has been relying on a private API which has been removed. https://github.com/elixir-lang/elixir/commit/9e40b8f786625b2f036ce9c2467cd0a8ade35ce6#diff-74de88d06654c488a26aac22a20602d5L292

**Comment by lua-net** [comment]
I have the same problem, I had to downgrade the elixir to 1.9.4 =/

**Comment by progsmile** [comment]
Also having this issue on `1.10.0` may try to fix

**Comment by lessless** [comment]
@inooid , @sneako  @lua-net  folks, did you have a chance to try fix by @progsmile https://github.com/edeliver/edeliver/pull/326 ?

**Comment by progsmile** [comment]
@lessless I've tried within fork, it works :)

**Comment by lessless** [comment]
New version is available as v1.8.0 https://hex.pm/packages/edeliver/1.8.0
https://github.com/edeliver/edeliver/blob/master/CHANGELOG.md

thanks again @progsmile
